### PR TITLE
🐛 fix: GitHub rework detection no longer filters out PR owner's comments

### PR DESCRIPTION
**Pull request by Pushedskydiver**
## Summary

- Replace `excludeAuthor` (author-based filtering) with `[clancy]` content-based filtering in GitHub rework detection
- Clancy's automated `[clancy] Rework pushed...` comments are still excluded from triggering rework
- User's own `Rework:` comments now pass through even when using the same GitHub token as Clancy
- Fixes infinite loop on GitHub PRs reported by Jamie

## Root cause

`checkGitHubPrReviewState()` used `excludeAuthor` (resolved from `GITHUB_TOKEN` via `resolveUsername`) to filter comments. Since Clancy uses the user's token, the PR author and Clancy are the same GitHub user — so the user's legitimate `Rework:` comments were silently discarded.

## Fix

Filter by content prefix instead of author. Clancy already prefixes all automated comments with `[clancy]` (see `buildReworkComment()` in rework.ts). Comments starting with `[clancy]` are filtered out; everything else passes through.

GitLab/Bitbucket verified — not affected (they have no author filtering, and `[clancy]` prefix doesn't match `Rework:` detection).

## Test plan

- [x] `npm test` — 1220 unit tests passing (3 new)
- [x] `npm run test:integration` — 112 integration tests passing
- [x] `npm run typecheck` — clean
- [x] `npm run lint` — clean
- [x] DA review — verified GitLab/Bitbucket not affected, no edge case gaps

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/scripts/shared/pull-request/github/github.ts [hunk 1]
```diff
@@ -81,12 +80,14 @@ export async function checkPrReviewState(
...
-      ? rawConvo.filter((c) => c.user?.login !== excludeAuthor)
-      : rawConvo;
+    // Filter out Clancy's own automated comments (prefixed with [clancy])
+    // to prevent self-triggering rework loops. User comments — including
+    // from the same GitHub account — pass through.
+    const isClancyComment = (body?: string | null) =>
+      body?.trimStart().startsWith('[clancy]') ?? false;
+
+    const inlineComments = rawInline.filter((c) => !isClancyComment(c.body));
+    const convoComments = rawConvo.filter((c) => !isClancyComment(c.body));
```
[the remark this answers is not in the document]
Comment (Pushedskydiver) [reply src/scripts/shared/pull-request/github/github.ts]: Fixed in 97ec725 — updated JSDoc for both `checkPrReviewState` and `fetchPrReviewComments` to document `[clancy]`-prefixed comment exclusion.
